nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/generic_error_code_condition.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__GENERIC_ERROR_CODE_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__GENERIC_ERROR_CODE_CONDITION_HPP_
+
+#include <string>
+#include <memory>
+#include <vector>
+#include <set>
+
+#include "rclcpp/rclcpp.hpp"
+#include "behaviortree_cpp_v3/condition_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class GenericErrorCodeCondition : public BT::ConditionNode
+{
+public:
+  GenericErrorCodeCondition(
+    const std::string & condition_name,
+    const BT::NodeConfiguration & conf)
+  : BT::ConditionNode(condition_name, conf)
+  {}
+
+  GenericErrorCodeCondition() = delete;
+
+  BT::NodeStatus tick() override
+  {
+    getInput<std::set<int>>("current_error_codes", current_error_codes_);
+
+    for (const auto & error_code_to_check : error_codes_to_check_) {
```

Comment:
I can add this back in pretty easily.

Reply:
Yeah, sorry, I probably misinterpreted the ports thing before. there's 1 input port and N settings for the different codes of interest. 

Though, maybe its better than passing a single error code source (e.g. `smooth_path_error_code` it should be a vector as well so it can service multiple. As the generic base class, that might be useful.